server/services/caldav/lib/calendar/calendar.syncUser.js

```diff
@@ -0,0 +1,54 @@
+const logger = require('../../../../utils/logger');
+
+/**
+ * @description Start user sync.
+ * @param {Object} userId - Gladys user to connect & sync.
+ * @returns {Promise} Resolving.
+ * @example
+ * syncUser(user.id)
+ */
+async function syncUser(userId) {
+  const account = await this.connect(userId);
+
+  // Clean all dav Calendar for full Sync
+  let gladysCalendars = await this.gladys.calendar.get(userId);
+  let davCalendars = gladysCalendars.filter((calendar) => calendar.service_id === this.serviceId);
+
+  await Promise.all(
+    davCalendars.map(async (calendar) => {
+      const events = await this.gladys.calendar.getEvents(userId, {
+        calendarId: calendar.id,
+      });
+
+      await Promise.all(events.map((event) => this.gladys.calendar.destroyEvent(event.selector)));
+
+      return this.gladys.calendar.destroy(calendar.selector);
```

Comment:
If I understand well, you destroy/re-create all events at each sync?

I see two problems with that:

- During the sync, Gladys is in a "buggy" state. If the user asks Gladys something, or if a scene runs during the sync,  Gladys will not be able to work and it will break, because the DB will be empty. As the sync happens every 30 minutes, it's something which will happen quite frequently.
- Unique ID are re-generated at each sync, meaning it can break things: an external script setup by the user, a scene which rely on an event

Why didn't you choose to update existing events?

Reply:
Yes, I choose to do it like that. In my first tests (when I develop the v3 module), I've been having problems by updating existing events when user choose to delete an event and he re-sync Gladys with caldav. I'm still gonna retry incremental synchronisation, we want the best solution.